### PR TITLE
[lit-next] Squelches most test error logging to make output less spammy.

### LINKED ISSUE #1556: [lit-next] Ensure tests do not trigger dev_mode warnings unless those warnings are being tested
## Description
Right now a bunch of tests trigger dev mode warnings, making the logs hard to parse. Ideally the tests do not trigger warnings unless these warnings are being tested.

## Acceptance criteria
PR landed.

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #1556.

### CODE REVIEW

**Review by kevinpschaaf — APPROVED** [review]
LGTM, nice & easy!

File: packages/tests/web-test-runner.config.js [hunk 1]
```diff
@@ -188,24 +188,11 @@ export default {
...
     }),
   ],
-  filterBrowserLogs: ({args}) => {
-    if (
-      mode === 'dev' &&
-      typeof args[0] === 'string' &&
-      args[0].includes('in dev mode')
-    ) {
-      if (!seenDevModeLogs.has(args[0])) {
-        seenDevModeLogs.add(args[0]);
```
Comment (kevinpschaaf) [inline packages/tests/web-test-runner.config.js:198]: Oh sorry, please remove `seenDevModeLogs`, no longer referenced.

Comment (kevinpschaaf) [reply packages/tests/web-test-runner.config.js:198]: Actually, what is the behavior now w.r.t. the dev logs?

Comment (sorvell) [reply packages/tests/web-test-runner.config.js:198]: They do not display. When running the tests, there is still a console log saying that you're using development or production builds.

**Review by kevinpschaaf — APPROVED** [review]
